var/spack/repos/builtin/packages/fastqc/package.py

```diff
@@ -29,8 +31,12 @@ def install(self, spec, prefix):
         mkdir(prefix.bin)
         mkdir(prefix.lib)
         install("fastqc", prefix.bin)
-        for j in ["cisd-jhdf5.jar", "jbzip2-0.9.jar", "sam-1.103.jar"]:
-            install(j, prefix.lib)
+        if self.spec.satisfies("@:0.11.9"):
+            for j in ["cisd-jhdf5.jar", "jbzip2-0.9.jar", "sam-1.103.jar"]:
+                install(j, prefix.lib)
+        else:
+            for j in ["cisd-jhdf5.jar", "jbzip2-0.9.jar", "htsjdk.jar"]:
+                install(j, prefix.lib)
```

Comment:
```suggestion
        install("cisd-jhdf5.jar", prefix.lib)
        install("jbzip2-0.9.jar", prefix.lib)
        
        if self.spec.satisfies("@:0.11.9"):
                install("sam-1.103.jar", prefix.lib)
        else:
                install("htsjdk.jar", prefix.lib)
```

This isn't a hard requirement, but I typically opt for making packages easier to read when possible instead of going for maximum flexibility in the future. Otherwise this PR looks great!